### PR TITLE
Add API for runtime remeshing

### PR DESCRIPTION
**Pull request by fsimonis**
This PR adds the API for runtime remeshing to the python bindings.

The feature is available in the develop version of preCICE https://github.com/precice/precice/pull/2070

As the limitations are still changing, I refer the user to the online documentation.

### CODE REVIEW

File: cyprecice/cyprecice.pyx [hunk 1]
```diff
@@ -710,6 +710,42 @@ cdef class Participant:
...
+        mesh_name : str
+            Name of the mesh to reset.
+
+        Notes
+        -----
+        This function is still experimental.
+        Please refer to the documentation on how to enable and use it.
+
+        Previous calls:
+            advance() been called
```
Comment (uekerman) [inline cyprecice/cyprecice.pyx:731]:
```suggestion
            advance() has been called
```

File: cyprecice/cyprecice.pyx [hunk 2]
```diff
@@ -710,6 +710,42 @@ cdef class Participant:
...
+
+        Examples
+        --------
+        Set mesh vertices for a 2D problem with 5 mesh vertices.
```
Comment (IshaanDesai) [inline cyprecice/cyprecice.pyx:735]:
```suggestion
        Reset a mesh with 5 vertices to have 3 vertices.
```

File: cyprecice/cyprecice.pyx [hunk 3]
```diff
@@ -710,6 +710,42 @@ cdef class Participant:
...
+        >>> vertex_ids = participant.set_mesh_vertices(mesh_name, positions)
+        >>> # later in the coupling loop
+        >>> if remeshing_required():
+        >>>     participant.reset_mesh(mesh_name)
+        >>>     positions = np.array([[1, 1], [3, 3], [5, 5]])
+        >>>     vertex_ids = participant.set_mesh_vertices(mesh_name, positions)
+        """
+
+        self.thisptr.resetMesh (convert(mesh_name))
+
```
Comment (IshaanDesai) [inline cyprecice/cyprecice.pyx:748]: Shouldn't there be a `return` statement returning the new vertex IDs?

File: test/Participant.cpp [hunk 4]
```diff
@@ -237,6 +237,11 @@ void Participant::setMeshTetrahedra
     precice::span<const precice::VertexID> vertices)
 {}
 
+void Participant::resetMesh
+(
+    precice::string_view meshName)
+{}
```
Comment (IshaanDesai) [inline test/Participant.cpp:243]: If such a mock function exists, can you please add a unit test?

File: cyprecice/cyprecice.pyx (hunk 3 above)
Comment (fsimonis) [reply cyprecice/cyprecice.pyx:748]: Do you mean in the example?

The `resetMesh()` API only empties the mesh. There is no return value.

File: test/Participant.cpp (hunk 4 above)
Comment (fsimonis) [reply test/Participant.cpp:243]: I added a test without assertions. Is this fine?

Comment (IshaanDesai) [reply test/Participant.cpp:243]: Looks good :+1:

**Review by IshaanDesai — APPROVED** [review]
Ready to merge.

**Pull request merged**